explainer.md

```diff
@@ -0,0 +1,459 @@
+* When the callback returns, the new DOM is considered ready for first render. This starts step 5 onwards to create new pseudo elements and start animations.
+
+## Additional Script APIs
+The following example shows how developers can configure the transition in script for an MPA.
+
+### Old Document
+```js
+addEventListener("navigate", (event) => {
+  // Add sharedId attribute to elements to offer for the transition
+  // based on current document state.
+  document.querySelector(".header").sharedId="header-id";
+  
+  // setData can be used to pass opaque contextual information to the
+  // new page. The argument type is |any|.
+  document.documentTransition.setData({ version: 123 });
+});
+```
+
+### New Document
+```js
+requestAnimationFrame(() => {
+  let pendingTransition = document.documentTransition.getPendingTransition();
+  if (pendingTransition.getData().version !== 123)
+    return;
+  
+  // |offeredTransitionElements| provides a list of objects to access state
+  // saved from the old page.
+  let oldHeader = pendingTransition.offeredTransitionItems.get("header-id");
+  if (oldHeader) {
+    // Add sharedId attribute to elements animated in the new DOM.
+    document.querySelector(".header").sharedId="header-id";
+    
+    // Query the style information saved from the old page.
+    let oldHeaderStyle = oldHeader.getContainerComputedStyle();
+    
+    // The pseudo elements for each shared element are associated with the root element.
+    // The existing [pseudoElement](https://drafts.csswg.org/web-animations-1/#dom-keyframeeffect-pseudoelement) option can be used
+    // to target them with Web Animations API.
+    document.documentElement.animate(
+      [{ width: oldHeaderStyle.width,
+        height: oldHeaderStyle.height,
+        transform: oldHeaderStyle.transform }],
+      { duration: 1000,
+        pseudoElement: '::shared-container(#header-id)' });
+  }
+});
+```
+
+# Alternatives Considered
+## Heirarchical Properties
+This proposal disallows a shared element to be nested inside another shared element. The restriction avoids the need to preserve the hierarchy of the shared elements and associated properties (transform, clip, effects inherited by descendents) when creating pseudo elements. This is a consideration for future iterations of the feature.
+
+## Container/Child Split
+One consideration was to render each shared element using a replaced element directly instead of creating a container element. The motivation behind this split is to provide a stacking context to cross-fade the content of old and new shared elements. This was necessary to ensure blending identical pixels is a no-op using [plus-lighter](https://drafts.fxtf.org/compositing/#porterduffcompositingoperators_plus_lighter) blending. While same-origin transitions could work around this, it enables future extensibility for cross-origin transitions where cross-fading identical images would be common.
```

Comment:
```suggestion
One consideration is to render each shared element using a replaced element directly instead of creating a container element. The motivation behind this split is to provide a stacking context to cross-fade the content of old and new shared elements. This was necessary to ensure blending identical pixels is a no-op using [plus-lighter](https://drafts.fxtf.org/compositing/#porterduffcompositingoperators_plus_lighter) blending. While same-origin transitions could work around this, it enables future extensibility for cross-origin transitions where cross-fading identical images would be common.
```